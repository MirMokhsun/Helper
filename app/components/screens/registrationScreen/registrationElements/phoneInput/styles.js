import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    conteiner: {
        height: 55,
        flexDirection: 'row',
        width: 290,
        marginBottom: 5,
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    phoneCodeWrapper: {
        height: 35,
        width: 50,
        marginBottom: 15,
        borderColor: '#1459D2',
        borderBottomWidth: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    text: {
        fontSize: 16,
        fontFamily: 'Roboto-Light',
        color: '#1459D2',
    }
});