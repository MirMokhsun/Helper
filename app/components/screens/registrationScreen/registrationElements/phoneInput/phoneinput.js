import React, { PureComponent } from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles';
import { CommonInput } from '../../../../commonComponents/commonInput/commonInput';

export class PhoneInput extends PureComponent {
    constructor(props) {
        super(props)
    }

    render() {
        const { currentLanguageName, phcode, placeholder, errorText, filter, action, isValid, maxWidth, value } = this.props;
        return (
            <View style={styles.conteiner}>
                <View style={styles.phoneCodeWrapper}>
                    <Text style={styles.text}>+{phcode}</Text>
                </View>
                <CommonInput
                    value = { value }
                    keyboardType='numeric'
                    maxWidth={maxWidth}
                    placeholder={placeholder}
                    errorText={errorText}
                    filter={filter}
                    action={action}
                    isValid={isValid}
                    testID="InputPhoneRegistrationID"
                    textErrorID="PhoneErrorRegostrationID"
                    currentLanguageName={currentLanguageName}
                />
            </View>
        )
    }
}

export default PhoneInput;