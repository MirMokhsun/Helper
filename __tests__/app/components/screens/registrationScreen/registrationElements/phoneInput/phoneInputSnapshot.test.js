import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { PhoneInput } from '../../../../../../../app/components/screens/registrationScreen/registrationElements/phoneInput/phoneinput';

it('create snapshot PhoneInput', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<PhoneInput
        phcode={380}
        maxWidth={240}
        placeholder={'phoneNumber'}
        errorText={'phoneNumber'}
        action={() => { }}
        isValid={true}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});