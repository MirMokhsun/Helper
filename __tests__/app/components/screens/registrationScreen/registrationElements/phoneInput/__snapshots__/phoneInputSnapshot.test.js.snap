// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot PhoneInput 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "space-between",
      "marginBottom": 5,
      "width": 290,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#1459D2",
        "height": 35,
        "justifyContent": "center",
        "marginBottom": 15,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1459D2",
          "fontFamily": "Roboto-Light",
          "fontSize": 16,
        }
      }
    >
      +
      380
    </Text>
  </View>
  <CommonInput
    action={[Function]}
    errorText="phoneNumber"
    isValid={true}
    keyboardType="numeric"
    maxWidth={240}
    placeholder="phoneNumber"
    testID="InputPhoneRegistrationID"
    textErrorID="PhoneErrorRegostrationID"
  />
</View>
`;
